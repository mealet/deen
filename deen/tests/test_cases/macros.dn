fn main() i32 {
  // print!(LITERAL_FORMAT, ...) void
  print!("print macro\n");

  // println!(LITERAL_FORMAT, ...) void
  println!("println macro");

  // format!(LITERAL_FORMAT, ...) *char
  let greet: *char = format!("Hello, {}!", "World");
  println!("{}", greet);

  // sizeof!(EXPRESSION / BASIC TYPE) usize
  println!("{}", sizeof!(i32));
  println!("{}", sizeof!(i64));

  // cast!(EXPRESSION, TYPE) TYPE
  println!("{}", cast!(123.123, i32));

  // panic!(LITERAL_FORMAT, ...) void
  // EXITING WITH CODE 1
  panic!("Panic test, 2 + 2 = {}", 2 + 2);

  return 0;
}
