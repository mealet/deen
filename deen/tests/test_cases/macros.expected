@! 1
print macro formatted
println macro formatted
Hello, World!
4
8
123

* Runtime Panic at `macros.dn` <line 21>
Panic test, 2 + 2 = 4
